File: src/my_robot_tutorials/scripts/robot_news_radio_transmitter.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String

if __name__ == '__main__':

	# We give our node a name, which is the same
	# as the file name, and it's ok in python scropts
	# the anonymous argument is optional, but it gives the
	# possibility that 2 nodes with the same name exist
	# because it adds some random numbers at the end of its name
	rospy.init_node('robot_news_radio_transmitter', anonymous=True)

	# We create a publisher using the Publisher class 
	# inside rospy. 1st we give a name to the topic and
	# then a message type. We will use the message types
	# provided by the std_msgs dependancy that we used
	# when we created the package (see ROS Wiki).
	# We will use String type that contains just string
	# data.
	# Last thing we need to add is a queue size (~buffer)
	pub = rospy.Publisher("/robot_news_radio", String, queue_size=10)

	# We define a rate of 2Hz
	rate = rospy.Rate(2)

	# Now let's publish some data
	while not rospy.is_shutdown():
		msg = String()
		# From ROS Wiki: in String message we have data
		msg.data = "Hi, this is J from the Robot News Radio !"
		# We publish the message
		pub.publish(msg)
		rate.sleep()

	# To ensure we see that the node was stopped when we make 
	# rosnode kill, we log a message after the loop
	rospy.loginfo("Node was stopped")

File: src/my_robot_tutorials/src/robot_news_radio_transmitter.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int	main(int argc, char **argv)
{
	// We initialize the node with its name, and we give an
	// optional argument to make it anonymous, this means
	// a set of random numbers will be added at the end
	// of the name when the new node is run avoiding naming 
	// conflict and so allowing several versions of the same 
	// node to work at the same time
	ros::init(argc, argv, "robot_news_radio_transmitter", \
		ros::init_options::AnonymousName);
	ros::NodeHandle nh;

	// We create a ros Publisher, and the way to create it is
	// through node handler. We need to give it the message type,
	// the topic name and the queue size
	ros::Publisher pub = nh.advertise<std_msgs::String>("/robot_news_radio", 10);

	// We create the rate at 3Hz
	ros::Rate rate(3);

	while (ros::ok())
	{
		// We create the message
		std_msgs::String msg;
		// Wee add the data field
		msg.data = "Hi, this is LLei from the Robot News Radio";
		// We pulish the message
		pub.publish(msg);
		rate.sleep();
	}
}
